test/promise/simple.js

```diff
@@ -49,4 +49,48 @@ describe('Promise', function () {
       Promise.all.call(StealingPromiseConstructor, iterable);
     });
   });
+
+  specify('resolve with a thenable calls it once', function () {
```

Comment:
No, mocha supports returning a Promise as the result of a test.  This is a much more convenient mechanism for async tests, and fixes bugs in your original implementation.